File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F14.P3.go
package main

import (
	"fmt"

	"probe/shapes"
)

// BEGIN PROBE F14.P3
type Tri struct{ b, h float64 }

func (t Tri) Area() float64 { return 0.5 * t.b * t.h }

func triArea() float64 {
	var s shapes.Shape = Tri{3.0, 4.0}
	return s.Area()
}

// END PROBE F14.P3

func main() {
	fmt.Println(triArea())
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/typescript/F14.P3.ts
import type { Shape } from "./shapes.js"

// BEGIN PROBE F14.P3
class Tri implements Shape {
constructor(private b: number, private h: number) {}
area(): number {
return 0.5 * this.b * this.h
}
}

function triArea(): number {
const s: Shape = new Tri(3.0, 4.0)
return s.area()
}
// END PROBE F14.P3

console.log(triArea())
